File: app/src/main/java/com/theflexproject/thunder/fragments/BaseFragment.java
package com.theflexproject.thunder.fragments;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import com.theflexproject.thunder.model.MediaItem;

public abstract class BaseFragment extends Fragment {

    protected FragmentActivity mActivity;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);

        if (context instanceof Activity){
            mActivity = (FragmentActivity) context;
        }
    }

    protected void onNewIntent(Intent intent) {
    }

}

File: app/src/main/java/com/theflexproject/thunder/fragments/BaseFragment.kt
package com.theflexproject.thunder.fragments

import android.app.Activity
import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

abstract class BaseFragment : Fragment() {

    @JvmField
    protected var mActivity: FragmentActivity? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)

        if (context is Activity) {
            mActivity = context as FragmentActivity
        }
    }
}
